### PR TITLE
Pull request - changes in Chemiscope.py and 1_LM

### PR DESCRIPTION
**Pull request by hurricane642**

### CODE REVIEW

**Comment by rosecers** [comment]
@hurricane642 can you re-commit stripping your notebooks? It's very hard to go over the diff otherwise.

You should be able to use [nbstripout](https://pypi.org/project/nbstripout/).

Also, why is CSD-1000R.json changed? that should be static.

File: notebooks/0_Foreword.ipynb
Comment (rosecers) [inline notebooks/0_Foreword.ipynb]: Make sure to remove any irrelevant diffs like these.

File: notebooks/2_PrincipalCovariatesRegression.ipynb
Comment (rosecers) [inline notebooks/2_PrincipalCovariatesRegression.ipynb]: Same comment as above.
```suggestion
   "version": "3.6.9"
```

File: notebooks/3_KernelMethods.ipynb
Comment (rosecers) [inline notebooks/3_KernelMethods.ipynb]: Same comment as above.
```suggestion
   "version": "3.6.9"
```

**Review by rosecers — CHANGES_REQUESTED** [review]
It is also useful to use a linter to format your code. At the very least with python code, [pyflakes](https://pypi.org/project/pyflakes/) and [autopep8 ](https://pypi.org/project/autopep8/) are a good place to start. There are a lot of formatting changes in the diff that should be remedied.

File: notebooks/1_LinearMethods.ipynb
Comment (rosecers) [inline notebooks/1_LinearMethods.ipynb]: Why this change?

File: notebooks/1_LinearMethods.ipynb
Comment (rosecers) [inline notebooks/1_LinearMethods.ipynb]: Why this change?

File: notebooks/0_Foreword.ipynb
Comment (rosecers) [inline notebooks/0_Foreword.ipynb:55]: This is not necessary

File: notebooks/1_LinearMethods.ipynb
Comment (rosecers) [inline notebooks/1_LinearMethods.ipynb]: please omit this line from the PR.

File: notebooks/1_LinearMethods.ipynb
Comment (rosecers) [inline notebooks/1_LinearMethods.ipynb]: @Luthaf might be interested in:
https://stackoverflow.com/questions/17097985/dict-vs-in-python-which-is-better/17098017

Comment (rosecers) [inline notebooks/1_LinearMethods.ipynb]:
```suggestion
    "    meta={'name':'CSD-1000R'},\n",
```

File: notebooks/3_KernelMethods.ipynb
Comment (rosecers) [inline notebooks/3_KernelMethods.ipynb]: Why this change?

File: notebooks/X_ImportingData.ipynb
Comment (rosecers) [inline notebooks/X_ImportingData.ipynb]: please omit this from the PR

File: utilities/chemiscope.py
Comment (rosecers) [inline utilities/chemiscope.py:69]: Was this added by a formatter or manually?

File: utilities/chemiscope.py
Comment (Luthaf) [inline utilities/chemiscope.py]: why change from None to an empty string / empty lists? chemiscope should be able to deal with this

File: utilities/chemiscope.py
Comment (rosecers) [inline utilities/chemiscope.py:98]: @Luthaf will this prevent a necessary error message?

**Review by rosecers — CHANGES_REQUESTED** [review]

File: utilities/chemiscope.py
Comment (hurricane642) [reply utilities/chemiscope.py]: I tried to start chemiscope with file, where "description".."references" were None, and it returned a mistake. So I changed it in this way, now it works.

File: notebooks/1_LinearMethods.ipynb
Comment (Luthaf) [reply notebooks/1_LinearMethods.ipynb]: I tend to prefer `{}` as well, you can have keys which are not valid identifiers: `{"a long key": 3}` =)

File: utilities/chemiscope.py
Comment (hurricane642) [reply utilities/chemiscope.py:69]: This was added by a formatter

File: utilities/chemiscope.py
Comment (Luthaf) [reply utilities/chemiscope.py]: ok, I would consider this a bug with chemiscope =) Could you share such a file with me (on slack or here) so that I can take a look?

File: utilities/chemiscope.py
Comment (Luthaf) [reply utilities/chemiscope.py:98]: No, but this is adding data where there is none, and we loose the distinction between "there is no reference information in this file" and "there are no references associated with this file", which I would like to preserve if possible. 

Chemiscope should be fine with `undefined` here, my guess with this issue is that the Python json module transforms `None` into `null`, which is different from `undefined`. There should be a way of dealing with this while keeping the None/`[]` separation.

File: utilities/chemiscope.py
Comment (hurricane642) [reply utilities/chemiscope.py:97]: Yes, what file do you need?

Comment (hurricane642) [reply utilities/chemiscope.py]: Yes, what file fo you need - notebook or a table?

Comment (Luthaf) [reply utilities/chemiscope.py]: The generated json file that gives you an error when loaded on chemiscope!

File: notebooks/1_LinearMethods.ipynb
Comment (rosecers) [inline notebooks/1_LinearMethods.ipynb]:
```suggestion
```

File: notebooks/4_SparseKernelMethods.ipynb
Comment (rosecers) [inline notebooks/4_SparseKernelMethods.ipynb]: Please delete this empty cell.

File: notebooks/4_SparseKernelMethods.ipynb
Comment (rosecers) [inline notebooks/4_SparseKernelMethods.ipynb]: Please remove this line from the PR.

File: notebooks/5_CUR.ipynb
Comment (rosecers) [inline notebooks/5_CUR.ipynb]:
```suggestion
    "from utilities.classes import PCA, LR\n",
```

File: notebooks/4_SparseKernelMethods.ipynb
Comment (rosecers) [inline notebooks/4_SparseKernelMethods.ipynb]:
```suggestion
   "version": "3.6.9"
```

File: utilities/chemiscope.py
Comment (rosecers) [inline utilities/chemiscope.py:107]: Do not delete this line.

File: utilities/chemiscope.py
Comment (rosecers) [inline utilities/chemiscope.py:103]: Do not delete this line.

File: notebooks/4_SparseKernelMethods.ipynb
Comment (rosecers) [inline notebooks/4_SparseKernelMethods.ipynb]: Please do not delete this line.

File: utilities/chemiscope.py
Comment (rosecers) [inline utilities/chemiscope.py:97]: @Luthaf might it be better like this?

```suggestion
            key: meta.get(key, None) for key in AUTHORIZED_KEYS
```

- This will also reduce loop time if there are ~infinite keys in meta.
- Also might suggest `AUTHORIZED_KEYS` --> `REQUIRED_KEYS`

**Review by rosecers — COMMENTED** [review]
@hurricane642 Provided the requested changes are made, I see no reason not to merge this into master once @Luthaf approves the changes to `chemiscope.py`.

File: utilities/chemiscope.py
Comment (Luthaf) [reply utilities/chemiscope.py:97]: no, the whole issue comes frome using `None`, which is translated into `null` by Python's json module, which is different from `undefined` on the JS side. 

> This will also reduce loop time if there are ~infinite keys in meta.

I think having a very large number of invalid keys in meta is not something we want to optimize for  =)

> Also might suggest AUTHORIZED_KEYS --> REQUIRED_KEYS

Only `name` is required, the other keys are not.

Comment (rosecers) [reply utilities/chemiscope.py:97]: Bah, I had so many plans for infinite keys in meta. @hurricane642 let's remove this change from this PR and circle back at another time.

Comment (Luthaf) [reply utilities/chemiscope.py:97]: I provided the above code (`key: value for key, value in meta.items() if key in AUTHORIZED_KEYS`) to @hurricane642 to fix the issue he's been seeing with the previous code (https://github.com/cosmo-epfl/kernel-tutorials/pull/17#discussion_r497389919), so the current state of the code is the right one here 😃

File: notebooks/5_CUR.ipynb
Comment (rosecers) [inline notebooks/5_CUR.ipynb]:
```suggestion
   "version": "3.6.8"
```

**Review by rosecers — APPROVED** [review]

**Pull request merged**
